kv/util/util_test.go

```diff
@@ -222,6 +244,14 @@ func TestWatchAndUpdateInt64(t *testing.T) {
 			break
 		}
 	}
+
+	// Updates should not be applied after the watch is closed.
+	watch.Close()
```

Comment:
could you ensure we're not leaking goroutines (use something like [leaktest](https://github.com/fortytw2/leaktest) )